alfred.c

```diff
@@ -109,91 +102,143 @@ alfred_exec (lua_State *ls, const char *script)
 }
 
 static bool
-node_changed (const char *path, const char *value)
+watch_node_changed (const char *path, const char *value)
 {
     char *_path = NULL;
-    alfred_t *alfred = NULL;
+    GList *matches = NULL;
+    GList *node = NULL;
+    GList *script = NULL;
+    GList *scripts = NULL;
     bool ret = false;
+    cb_info_t *cb = NULL;
 
     assert (path);
     assert (alfred_inst);
 
-    /* Find a matching action */
-    _path = g_strdup (path);
-    alfred = g_hash_table_lookup (alfred_inst->actions, _path);
-    while (alfred == NULL && strlen (_path) > 1)
-    {
-        *(strrchr (_path, '/')) = '\0';
-        alfred = g_hash_table_lookup (alfred_inst->actions, _path);
-    }
+    _path = g_strdup_printf ("%s", path);
```

Comment:
g_strdup ... do we need this at all? Does cb_match stomp on the path?

Probably pass path into cb_match